core/src/main/java/org/frankframework/pipes/RegExPipe.java

```diff
@@ -122,20 +129,21 @@ public void configure() throws ConfigurationException {
 
 	@Override
 	public PipeRunResult doPipe(Message message, PipeLineSession session) throws PipeRunException {
-		String sInput;
 		if (Message.isEmpty(message)) {
-			sInput = "";
-		} else {
-			try {
-				sInput = message.asString();
-			} catch (IOException e) {
-				throw new PipeRunException(this, "cannot open stream", e);
-			}
-			if (sInput == null) {
-				throw new PipeRunException(this, "Input message is empty");
-			}
+			return new PipeRunResult(elseForward, null);
 		}
 
+		String sInput;
+		try {
+			sInput = message.asString();
+		} catch (IOException e) {
+			throw new PipeRunException(this, "cannot open stream", e);
+		}
+		if (StringUtils.isEmpty(sInput)) {
+			throw new PipeRunException(this, "Input message is empty");
+		}
```

Comment:
Throwing an exception on empty string here is inconsistent with returning a PipeRunResult with the elseForward on line 133.

A message is empty when it is `null`, or has size 0 (so if the message contains an empty string, this means the elseForward is selected).

If the size of the message is unknown at that point (because, for instance, it is an InputStream) the message is read and if now that InputStream results in an empty string, an exception is thrown instead of taking the elseFoward.

That means there is different behaviour on what is effectively the same input depending on the source representation.

I think that's wrong.

Also, you stated earlier that an empty string might be valid because you can configure a regex that matches the empty string. Now, that is not possible anymore, an empty string will never match.

Reply:
Agreed. I would propose the following contract:

null -> else
string -> try to match

Reply:
I think that there are several places where we equate `null` input with an empty string. So I wouldn't do that either.

Reply:
What would you suggest? Always going to the else branch when the input is null or empty is also acceptable to me. This would prevent a wildcard match on an empty string, but I don't think that there are any use cases for that.

Reply:
I suspect we have other ways to detect empty input so probably not many use-cases for that indeed.

Unless others think otherwise, let's always go to the else-forward on any empty input.